### ISSUE TITLE
perf(vulkan): bind lifetime-aware transient plans to native suballocated buffer views

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12223]
Parent campaign: #11572

Order: **8 of 16** — P1, converts an already-proven lifetime plan into real allocation reuse.

## For / Problem / Today / Better because / Witness

- **For:** Qwen3.8 Vulkan decode and sequence prefill.
- **Problem:** every logical transient still becomes a native buffer/memory allocation despite non-overlapping lifetimes.
- **Today:** `PlanGraphTransients` at `internal/compute/vulkan_transient_plan.go:71` is pure planning; native allocation begins at `internal/compute/vulkan_shim.cpp:335`.
- **Better because:** aligned views into bounded backing arenas can reuse memory across disjoint lifetimes and reduce allocation count/peak memory.
- **Witness:** live-range non-overlap proof, identical outputs, fewer `vkAllocateMemory` calls, bounded peak bytes, and physical load/step measurements.

## Evidence and borrow

Self-query finds only generic planning/benchmark cards; the native binding is absent. **PARTIAL**.

Adapt Modular's stack-reuse principle at `StackReuse.cpp@1c9fd2e03331f77d3a1034127cb3700b7fa43c02` under its repository license; local source already records this provenance. Supporting allocator study: `study_a8e74aeef35f4f79a52d2af0a1d1f6ef8c4830eabdf419db4ddf3a2052c08c1a`.

## Dependencies and fences

Depends on #11922 and the immutable arena leaf above. Distinct from #9834: this changes buffer ownership/views, not command-graph replay.

## Acceptance

- Bind planner offsets to native buffer views with correct alignment.
- Prove no simultaneously live tensors overlap.
- Ensure retirement waits for all descriptor users.
- Keep only with source-bound gfx1151 memory/latency improvement and exact logits.

First check: `go test ./internal/compute -run TestVulkanTransientPlanBindsSuballocatedViews -count=1`.

**Comment by anthony-chaudhary** [comment]
Blocked pending required prerequisites; no implementation started.

Read-only scoping receipt (2026-09-08):
- #11922 is OPEN, with no linked PR or resolving commit found.
- #12217 (the immutable Vulkan arena leaf referenced by this issue) is OPEN, with no linked PR or resolving commit found.
- At `88b7c473d66f5d400143975c56391dec5f62f764`, `PlanGraphTransients` remains planning-only, while native `Buffer` still owns one `VkDeviceMemory` and `allocBuffer` binds each buffer at offset 0. The requested `TestVulkanTransientPlanBindsSuballocatedViews` is not present yet.
- Exact-tree admission check was collision-free: `dos arbitrate --workspace . --lane compute --kind keyword --tree internal/compute/vulkan_transient_plan.go internal/compute/vulkan_shim.cpp --output json --explain` returned GO against 0 live leases. I did not acquire/journal a lease because dependency state forbids edits.

Next unblock condition: close #11922 and #12217 with their contract/witnesses landed, then re-arbitrate and implement the smallest native view-binding contract on the two stated files (plus the single focused test if the issue boundary permits it), followed by the required strix1 source/binary/device/model/workload receipt with `fallback_count=0`.

**Comment by anthony-chaudhary** [comment]
Operator dispatch override (2026-09-08): **start implementation now**. Open dependencies and missing physical-device receipts are coordination inputs, not stop conditions. Build against current `origin/main`; use the smallest issue-local adapter or explicit interface assumption where an upstream seam is not yet present; coordinate assumptions and collisions with adjacent workers through issue comments. A live overlapping exact-tree lease still forbids conflicting writes, but narrow to a tree-disjoint test, shader, contract, or adapter slice and make tangible progress instead of returning `BLOCKED`. Run software tests and guarded validation now; obtain the Strix hardware receipt when the implementation is ready. The `gated/hardware` label is removed until hardware is the actual remaining step.

**Comment by anthony-chaudhary** [comment]
Operator correction accepted; implementation resumed on current `origin/main`.

Issue-local binding assumptions for adjacent Vulkan workers:
- The caller queries `fvk_transient_buffer_alignment(bytes)` for every logical view and passes the maximum positive power-of-two alignment to both `PlanGraphTransients` and `BindVulkanTransientViews`.
- One device-local-only `TransientArena` backs the returned views. There is no host-visible or llama.cpp fallback; allocation/type/alignment/bounds mismatch returns failure for the caller to classify.
- Each logical tensor remains a distinct `VkBuffer` with descriptor offset zero; only its `vkBindBufferMemory` offset aliases the arena. Go-side binding validation refuses simultaneously-live or escaping byte overlap.
- `fvk_transient_arena_free` first fences an open batch; arena memory remains alive until every view has passed `fvk_free`, including views parked by recorded descriptor users.
- Physical-pool admission from #11922 is not inferred in this slice. The native allocation failure stays explicit so that interface can be reconciled when it lands.

Borrow disposition: **ADAPT** from Modular `KGEN/lib/Transforms/StackReuse.cpp@1c9fd2e03331f77d3a1034127cb3700b7fa43c02` (source event 2026-08-28, observed 2026-09-08, Apache-2.0 with LLVM exceptions). Reused principle: conservative escape/lifetime eligibility; Vulkan allocation and retirement code is fak-native. Self-query remains PARTIAL/ABSENT (`fak capabilities "Vulkan lifetime-aware transient suballocated buffer views"` returned no matching capability; raw source showed planning only).

Focused software witness is now green: `go test ./internal/compute -run TestVulkanTransientPlanBindsSuballocatedViews -count=1`.

**Comment by anthony-chaudhary** [comment]
Implemented the tree-disjoint software binding slice in `dd529f2d0faf9c8c661028460730e32ba8a420b1` (`perf(vulkan): add transient view bindings (#12223) (fak compute)`).

What landed:

- `BindVulkanTransientViews` maps a non-empty lifetime plan to one native arena allocation plus an aligned buffer view per logical value.
- The binding boundary fails closed on malformed mode/alignment/bounds/lifetimes, duplicate nodes, forward-bump overlap, escaping aliases, and simultaneously-live byte-range aliases.
- `TestVulkanTransientPlanBindsSuballocatedViews` proves disjoint values reuse one suballocation, simultaneous values do not, all views are aligned/in-bounds, and a corrupted simultaneous overlap is rejected.

Software witnesses:

- `go test ./internal/compute -run '^TestVulkanTransientPlanBindsSuballocatedViews$' -count=1` — PASS.
- `go vet ./internal/compute` — PASS.
- `fak-dev buildcheck --vet --mine internal/compute/vulkan_transient_plan.go --mine internal/compute/vulkan_transient_plan_test.go` — PASS.
- Managed `fak worktree worker land` — PASS: exact-path admission, prospective build, policy admission, whole-tree disambiguation, post-merge build, remote recovery read-back, and race-free trunk CAS.
- `fak validate --test-only ...` software test phase — PASS. Its automatic Strix phase refused source binding because the appliance checkout was on a different Git tip and this Windows host did not resolve the configured SSH identities; per the operator's hardware-later direction, that physical receipt remains pending.

The native shim stayed untouched to remain tree-disjoint from concurrent Vulkan work. Shared `main` publication is queued by `fak sync drain`; the commit is already published under recovery ref `refs/fak/worker-land/fak/dd529f2d0faf9c8c661028460730e32ba8a420b1`. Keeping this issue open for native descriptor-retirement integration and the Strix keep/reject receipt.
